client/client_test.go

```diff
@@ -735,13 +735,9 @@ func testFrontendImageNaming(t *testing.T, sb integration.Sandbox) {
 			require.Equal(t, exporterResponse["image.name"], imageName)
 
 			// check if we can pull (requires containerd)
-			var cdAddress string
-			if cd, ok := sb.(interface {
-				ContainerdAddress() string
-			}); !ok {
+			cdAddress := sb.ContainerdAddress()
+			if cdAddress == "" {
 				return
```

Comment:
Should return skip here so we would notice when something like this happens. Or if a test has meaningful parts before the return maybe log out that "containerd specific parts were skipped".

Reply:
Updated to have them call `t.Skip()`. For the tests w/ meaningful parts before, they will still be marked as Failed if any of their `require` assertions fail before the skip call.